x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/utils/utils.test.ts

```diff
@@ -1382,4 +1384,221 @@ describe('utils', () => {
+        fieldCaps: fieldCapsMock,
+      } as unknown as ElasticsearchClient;
+
+      internalEsClient = {
+        ilm: {
+          explainLifecycle: ilmExplainLifecycleMock,
+        },
+      } as unknown as ElasticsearchClient;
+
+      jest.clearAllMocks();
+    });
+
+    it('should return frozen indices when indices are confirmed as frozen', async () => {
+      const inputIndices = ['index-*', 'regular-index'];
+      const fieldCapsResponse: FieldCapsResponse = {
+        indices: ['partial-index-1', 'partial-index-2', 'regular-index'],
+        fields: {},
+      };
+      const ilmExplainLifecycleResponse: IlmExplainLifecycleResponse = {
+        indices: {
+          'partial-index-1': {
+            index: 'partial-index-1',
+            managed: true,
+            phase: 'frozen',
+          },
+          'partial-index-2': {
+            index: 'partial-index-2',
+            managed: true,
+            phase: 'frozen',
+          },
+        },
+      };
+
+      fieldCapsMock.mockResolvedValue(fieldCapsResponse);
+      ilmExplainLifecycleMock.mockResolvedValue(ilmExplainLifecycleResponse);
+
+      const params = {
+        inputIndices,
+        internalEsClient,
+        currentUserEsClient,
+        to: 'now',
+        from: 'now-1d',
+        primaryTimestamp: '@timestamp',
+        secondaryTimestamp: undefined,
+      };
+
+      const frozenIndices = await checkForFrozenIndices(params);
+
+      expect(frozenIndices).toEqual(['partial-index-1', 'partial-index-2']);
+      expect(fieldCapsMock).toHaveBeenNthCalledWith(1, {
+        index: inputIndices,
+        fields: ['_id'],
+        ignore_unavailable: true,
+        index_filter: expect.any(Object),
+      });
+      expect(ilmExplainLifecycleMock).toHaveBeenNthCalledWith(1, {
+        index: inputIndices.join(','),
+      });
+    });
+
+    it('should return an empty array when no frozen indices are found', async () => {
+      const inputIndices = ['regular-index-1', 'regular-index-2'];
+      const fieldCapsResponse: FieldCapsResponse = {
+        indices: ['regular-index-1', 'regular-index-2'],
+        fields: {},
+      };
+
+      fieldCapsMock.mockResolvedValue(fieldCapsResponse);
+
+      const params = {
+        inputIndices,
+        internalEsClient,
+        currentUserEsClient,
+        to: 'now',
+        from: 'now-1d',
+        primaryTimestamp: '@timestamp',
+        secondaryTimestamp: undefined,
+      };
+
+      const frozenIndices = await checkForFrozenIndices(params);
+
+      expect(frozenIndices).toEqual([]);
+
+      expect(fieldCapsMock).toHaveBeenCalledWith({
+        index: inputIndices,
+        fields: ['_id'],
+        ignore_unavailable: true,
+        index_filter: expect.any(Object),
+      });
+      expect(ilmExplainLifecycleMock).not.toHaveBeenCalled();
+    });
+
+    it('should return an empty array when there are indices whose name begins with "partial-", but they are not frozen', async () => {
```

Comment:
this would be a good test case to cover with an integration test as well

Reply:
I have included this case in the integration tests.